Dockerfile

```diff
@@ -0,0 +1,12 @@
+#Build stage
+FROM gradle:jdk17-corretto AS build
+WORKDIR /usr/app/
+COPY . .
+RUN gradle build -x test
+
```

Comment:
Create a Non-Root User in the Final Image
```
RUN useradd -m appuser
USER appuser
```
Security Best Practice: Running as a non-root user prevents privilege escalation in case of an attack